### PR TITLE
Add bigquery authorized resources

### PR DESCRIPTION
**Pull request by thinhha**
Add options to include bigquery authorized resources. I tried to also make the change backwards compatible with the existing approach to create authorized views via the `access` block.

This is my first PR to this repo. I don't fully understand how the tests work. Any feedback will be much welcomed.

### CODE REVIEW

**Comment by lcaggio** [comment]
Thanks for contributing and implementing such a great feature to have.

To fix the linting error, you can create the variables/output section with the tfdocs.py tool.

```
./tools/tfdoc.py modules/my-changed-module
```

For more details take a look at the CONTRIBUTING.md file.

File: modules/bigquery-dataset/README.md [hunk 1]
```diff
@@ -54,6 +54,70 @@ module "bigquery-dataset" {
 # tftest modules=1 resources=2 inventory=iam.yaml
 ```
 
+### Authorized Views, Datasets, and Routines
```
Comment (lcaggio) [inline modules/bigquery-dataset/README.md:57]: What about having a complete example with the raw resources and the authorized resources? Similar to the one in the resource [doc](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_dataset#example-usage---bigquery-dataset-authorized-dataset)?

File: modules/bigquery-dataset/outputs.tf [hunk 2]
```diff
@@ -27,7 +27,9 @@ output "dataset_id" {
     google_bigquery_dataset_access.group_by_email,
     google_bigquery_dataset_access.special_group,
     google_bigquery_dataset_access.user_by_email,
-    google_bigquery_dataset_access.views
+    google_bigquery_dataset_access.authorized_views,
```
Comment (lcaggio) [inline modules/bigquery-dataset/outputs.tf:30]: Alphabetical order.

File: modules/bigquery-dataset/outputs.tf [hunk 3]
```diff
@@ -39,7 +41,9 @@ output "id" {
...
```
Comment (lcaggio) [inline modules/bigquery-dataset/outputs.tf:44]: Alphabetical order.

File: modules/bigquery-dataset/outputs.tf [hunk 4]
```diff
@@ -51,7 +55,9 @@ output "self_link" {
...
```
Comment (lcaggio) [inline modules/bigquery-dataset/outputs.tf:58]: Alphabetical order.

File: modules/bigquery-dataset/README.md (hunk 1 above)
Comment (lcaggio) [reply modules/bigquery-dataset/README.md:57]: Probably a diagram may also help to understand.

**Review by lcaggio — COMMENTED** [review]
Thanks for the great addition. few comments.

File: modules/bigquery-dataset/variables.tf [hunk 5]
```diff
@@ -36,6 +36,35 @@ variable "access_identities" {
...
+    dataset_id = string,
+    routine_id = string
+  }))
+  default = []
+}
+
+variable "authorized_views" {
+  description = "An array of views to be authorized for the dataset."
+  type = list(object({
+    dataset_id = string,
```
[the remark this answers is not in the document]
Comment (lcaggio) [reply modules/bigquery-dataset/variables.tf:61]: That was my understanding, I would probably change `... for the dataset.` with `... on the dataset.`, in line with the auth datasets and routines.

File: modules/bigquery-dataset/README.md (hunk 1 above)
Comment (thinhha) [reply modules/bigquery-dataset/README.md:57]: I've added a more complete example in the latest PR

File: modules/bigquery-dataset/outputs.tf (hunk 2 above)
Comment (thinhha) [reply modules/bigquery-dataset/outputs.tf]: Great spot! Updated accordingly.

File: modules/bigquery-dataset/outputs.tf (hunk 3 above)
Comment (thinhha) [reply modules/bigquery-dataset/outputs.tf]: Great spot! Updated accordingly.

File: modules/bigquery-dataset/outputs.tf (hunk 4 above)
Comment (thinhha) [reply modules/bigquery-dataset/outputs.tf]: Great spot! Updated accordingly.

File: modules/bigquery-dataset/variables.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (thinhha) [reply modules/bigquery-dataset/variables.tf:61]: updated accordingly

**Comment by thinhha** [comment]
could you please take another look? Thank you @lcaggio

**Review by lcaggio — APPROVED** [review]

**Pull request merged**